### PR TITLE
Model tuning XGBoost

### PR DESCRIPTION
**Pull request by 007-SARANG**
- Previous model accuracy: ~95%
- Random Forest (with tuning): ~96% R²
- XGBoost (with tuning): ~97.3% R²
# This PR adds advanced hyperparameter tuning and model comparison for rice yield prediction. XGBoost outperformed previous models and is now saved for future use.

### CODE REVIEW

**Comment by nupurmadaan04** [comment]
@007-SARANG Great work Sarang , appreciated

**Comment by 007-SARANG** [comment]
@nupurmadaan04 Please add the gssoc25 label to this problem, as I’m not receiving any points on the leaderboard for this PR because of that.
Thank You
